### DISCUSSION TITLE
Bashly CLI framework - Retype-powered docs

### DISCUSSION THREAD

**Discussion by DannyBen** [retypeapp/retype#93] [Show and tell]
So I am the owner of the first post in this topic? Neat.

I would like to share my first Retype-powered documentations site, made for my Bash CLI framework named bashly.

https://bashly.dannyb.co

I have utilized many of the nice features in Retype, including tabs, panels, "edit this page", buttons, titled code blocks and reference links.

I hope someone else finds this documentation site useful, and perhaps even bashly itself

And for the Retype guys: You are an elite group of people. The product is top notch (and I have tried all the alternatives), and the team looks super awesome.

**Comment by geoffreymcgill** [comment]
Your Bashly docs look amazing! 

I really like your use of the `[!button]` with icon to link to the Examples. Looks great and cleanly highlights the feature.

<img width="224" alt="Screen Shot 2021-07-24 at 11 52 56 AM" src="https://user-images.githubusercontent.com/62210/126877255-b57d659b-1d19-43b0-8a27-48e79db2d291.png">

It's always so cool seeing well crafted documentation in a Retype site. My first thought when I open those sites is always... _oh, this looks so clean. Nice job!_, and forgetting that it's actually _powered by Retype_. It's happened a few times now.

Nice job. Looks like an interesting library, and you have excellent docs!

**Reply by DannyBen** [reply]
Why thank you!

> oh, this looks so clean. Nice job!

Exactly how I felt when I opened Retype. The retype.com site is a perfect showcase, and organized so well, that it inspires others to actually use these cool and clean widgets.

I have already deployed another Retype book, and working on the third now. It takes about 24 hours (gross) to convert from mdbook to retype (only because I am reshaping things into widgets and such).

EDIT:

Oh - and dont forget the matching collapsible "example panel" ([here](https://bashly.dannyb.co/configuration/command/)):

![image](https://user-images.githubusercontent.com/2405099/126877979-f0ae834a-4320-4fb1-adc5-29960f1a7351.png)

**Reply by geoffreymcgill** [reply]
> Oh - and dont forget the matching collapsible "example panel" ([here](https://bashly.dannyb.co/configuration/command/)):

ya, I really like your use of those Panels too. Your configuration provides a super clean UI for the content.

**Reply by geoffreymcgill** [reply]
> It takes about 24 hours (gross) to convert from mdbook to retype (only because I am reshaping things into widgets and such).

As you are upgrading from mdbook, I would love to hear about any manual steps you needed to make. Any configuration you needed to bring across. Any mdbook specific widgets or syntax within the `.md` that you needed to convert or adjust. Really anything that required a manual step during the upgrade.

Of course, minus any _reshaping_ of the content. Although, if there are specific scenarios where converting some specific would be helpful, then we can take a look at that too. We handle a lot of automatic conversion options when upgrading to Retype from GitBook, so those same processes can also handle upgrading from other systems.

Our goal is that if you run `retype watch` or `retype init` on an existing mdbook project, that Retype should be able to automatically run the upgrade process. The process should _just work_, have essentially all the same functionality, reproduce the navigation, and look awesome. 🕺 

Ideally, the upgrade should take <1 second and the result would be a near perfect re-implementation of site. That leaves you to focus on your content and revising to take advantage of all the extra features and components that Retype offers. That would be the sweet spot. 

Anything you can think of, even what you may think are trivial enhancements, would get me full attention.

**Reply by DannyBen** [reply]
> As you are upgrading from mdbook, I would love to hear about any manual steps

Sure. I will make notes and post in a new discussion.

> Our goal is that if you run retype watch or retype init on an existing mdbook project

That's an ambitious goal. I wouldn't expect retype CLI to do that for me. Things are different between books, and I accept the fact that it takes several hours to do so. Nevertheless, will keep that in mind when I compile my notes.

**Comment by geoffreymcgill** [comment]
> And for the Retype guys: You are an elite group of people. The product is top notch (and I have tried all the alternatives), and the team looks super awesome.

Woot woot! Thanks for all your interest in Retype. :100:

You've had some very quotable comments lately. Would you mind if we used some of your recent feedback and comments on the Retype website?

**Reply by DannyBen** [reply]
> You've had some very quotable comments lately. Would you mind if we used some of your recent feedback and comments on the Retype website?

Heh... and believe it or not, I hold back since I feel I am taking too much of your time...

You can use whatever you want - I dare you to find a bigger retype fan than I am. Consider me at your disposal for whatever I can do to help.
